watertap/unit_models/electrodialysis_0d.py

```diff
@@ -0,0 +1,938 @@
+            doc="Mass transfer term for the concentrate channel",
+        )
+        def eq_mass_transfer_term_concentrate(self, t, p, j):
+            return self.concentrate_channel.mass_transfer_term[t, p, j] == 0.5 * (
+                self.elec_migration_flux_in[t, p, j]
+                + self.elec_migration_flux_out[t, p, j]
+                + self.nonelec_flux_in[t, p, j]
+                + self.nonelec_flux_out[t, p, j]
+            ) * (self.cell_width * self.cell_length)
+
+        # Add isothermal condition
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Isothermal condition for the diluate channel",
+        )
+        def eq_isothermal_diluate(self, t):
+            return (
+                self.diluate_channel.properties_in[t].temperature
+                == self.diluate_channel.properties_out[t].temperature
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Isothermal condition for the concentrate channel",
+        )
+        def eq_isothermal_concentrate(self, t):
+            return (
+                self.concentrate_channel.properties_in[t].temperature
+                == self.concentrate_channel.properties_out[t].temperature
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Electrical power consumption of a stack",
+        )
+        def eq_power_electrical(self, t):
+            return self.power_electrical[t] == self.current[t] * self.voltage[t]
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Diluate_volume_flow_rate_specific electrical power consumption of a stack",
+        )
+        def eq_specific_power_electrical(self, t):
+            return (
+                pyunits.convert(
+                    self.specific_power_electrical[t],
+                    pyunits.watt * pyunits.second * pyunits.meter**-3,
+                )
+                * self.diluate_channel.properties_out[t].flow_vol_phase["Liq"]
+                == self.current[t] * self.voltage[t]
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Overall current efficiency evaluation",
+        )
+        def eq_current_efficiency(self, t):
+            return (
+                self.current_efficiency[t] * self.current[t]
+                == sum(
+                    self.diluate_channel.properties_in[t].flow_mol_phase_comp["Liq", j]
+                    * self.config.property_package.charge_comp[j]
+                    - self.diluate_channel.properties_out[t].flow_mol_phase_comp[
+                        "Liq", j
+                    ]
+                    * self.config.property_package.charge_comp[j]
+                    for j in self.config.property_package.cation_set
+                )
+                * Constants.faraday_constant
+            )
+
+    # initialize method
+    def initialize(
```

Comment:
```suggestion
    def initialize_build(
```
I believe newer version of IDAES will require the `initialize` function for custom unit models to be called `initialize_build`. It is basically the same function, but has some different args. Have a look at #516 for an example.